### PR TITLE
Use fog-aws, not all of fog

### PR DESCRIPTION
**Pull request by mlarraz**
Re-opening this from #53 because I still think it's pretty significant.

In short:
  * Landable has optional features that use Carrierwave
  * Carrierwave supports multiple S3 clients and doesn't explicitly depend on any of them
  * Fog has historically been the recommended S3 client for Carrierwave
  * Fog is a massive gem, which Landable only needs a small part of
  * Fog has extracted all its AWS features to [fog-aws](https://github.com/fog/fog-aws)
  * Fog itself [depends on fog-aws](https://github.com/fog/fog/blob/master/fog.gemspec#L56)
  * Carrierwave's [README now recommends fog-aws](https://github.com/carrierwaveuploader/carrierwave#using-amazon-s3)

tl;dr there is no good reason to keep explicitly depending on all of Fog, especially when many Landable users don't even use the Carrierwave functionality

### CODE REVIEW

**Comment by avavita** [comment]
I'm gonna close this and merge it into our develop branch for this iteration so we can test it alongside the rest of the next release. Thanks a lot!

**Comment by avavita** [comment]
eh nevermind I'll just update our dev.
